src/Confluent.Kafka/Producer.cs

```diff
@@ -667,7 +669,9 @@ internal Producer(ProducerBuilder<TKey, TValue> builder)
             Message<TKey, TValue> message,
             CancellationToken cancellationToken)
         {
-            Headers headers = message.Headers ?? new Headers();
+            Headers headers = message.Headers == null
+                ? message.Headers = new Headers()
+                : message.Headers;
```

Comment:
please keep changes focused to only those required (particularly where they're opinionated like this one, but also white space). this is important for navigating the code later in understanding what was changed when (git blame).

re: white space, it's good to remove it, yes, but i think the above consideration wins out, so i think the policy is to fix this only when making changes for some other reason.

Reply:
There is actually a logical difference, but you have to look really closely to see it 🔍 The previous version sets local `Headers headers` variable. The new version sets local `Headers headers` variable AND sets `message.Headers` prop to the same. That was needed for something else on the PR, but I don't remember exactly what at the moment.